File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    HIT = 1
    MISS = 5
    
    cache = []
    answer = 0
    
    if cacheSize == 0: return len(cities) * MISS

    for city in cities:
        city = city.lower()
        if city in cache:
            cache.remove(city)
            cache.append(city)
            answer += HIT
        else:
            if len(cache) == cacheSize:
                cache.pop(0)
            cache.append(city)
            answer += MISS
            
    return answer

File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        Deque<String> q = new ArrayDeque<>();

        for (String city : cities) {
            city = city.toUpperCase();
            if (q.contains(city)) {
                answer += 1;
                q.remove(city);
            } else {
                answer += 5;
            }
            q.addLast(city);
            while (q.size() > cacheSize) {
                q.removeFirst();
            }
        }
        return answer;
    }
}